torch/csrc/jit/interpreter.cpp

```diff
@@ -904,14 +908,24 @@ struct InterpreterStateImpl {
         // std::cout << "executing " << pc << ": ";
         // function->dumpInstruction(std::cout, pc);
         // std::cout << "\n";
-        auto & inst = instructions[pc];
-        loadTensorsFromRegisters(inst.inputs, stack);
-        pc += 1 + inst.callback(stack);
-        for(int i = inst.outputs.size - 1; i >= 0; i--) {
-          int reg = get(inst.outputs,i);
-          registers[reg] = pop(stack);
-          // std::cout << "pop reg[" << reg << "];\n" << registers[reg].pImpl << "\n";
+        size_t new_pc;
+        try {
+          auto & inst = instructions[pc];
+          loadTensorsFromRegisters(inst.inputs, stack);
+          new_pc = pc + 1 + inst.callback(stack);
+          for(int i = inst.outputs.size - 1; i >= 0; i--) {
+            int reg = get(inst.outputs,i);
+            registers[reg] = pop(stack);
+            // std::cout << "pop reg[" << reg << "];\n" << registers[reg].pImpl << "\n";
+          }
+        } catch(std::exception & e) {
+          if(pc >= instructions.size() || !instructions[pc].debug_location)
+            throw; // rethrow original exception
+          // throw a new exception with enhanced debugging information
+          instructions[pc].debug_location->wrapAndRethrowException(e, "operation failed in interpreter");
         }
+
+        pc = new_pc;
```

Comment:
nit: `new_pc` could be local to the try block - you can put this assignment as the last statement